From: Flavio Leitner <fleitner@redhat.com>
Date: Thu, 10 Dec 2009 17:04:17 -0500
Subject: [net] bnx2: fix frags index
Message-id: <20091210170417.GA25895@redhat.com>
Patchwork-id: 21852
O-Subject: [PATCH RHEL5.5] bnx2: fix frags index
Bugzilla: 546326
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Michal Schmidt <mschmidt@redhat.com>
RH-Acked-by: Jiri Olsa <jolsa@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>

RHBZ: bnx2: panic in bnx2_free_tx_skbs() because of wrong frags index
https://bugzilla.redhat.com/show_bug.cgi?id=546326

Description:

Signed-off-by: Don Zickus <dzickus@redhat.com>

diff --git a/drivers/net/bnx2.c b/drivers/net/bnx2.c
index 2ee87aa..dd2272f 100644
--- a/drivers/net/bnx2.c
+++ b/drivers/net/bnx2.c
@@ -5304,7 +5304,7 @@ bnx2_free_tx_skbs(struct bnx2 *bp)
 				tx_buf = &txr->tx_buf_ring[j + k + 1];
 				pci_unmap_page(bp->pdev,
 					pci_unmap_addr(tx_buf, mapping),
-					skb_shinfo(skb)->frags[j].size,
+					skb_shinfo(skb)->frags[k].size,
 					PCI_DMA_TODEVICE);
 			}
 			dev_kfree_skb(skb);
